### PR TITLE
updated gmsh2nek and exo2nek to create v004 mesh

### PR DESCRIPTION
**Pull request by yhaomin2007**

### CODE REVIEW

**Comment by dshaver14** [comment]
We should revise this. I would recommend always doing the non-right-hand check and just print the number of "suspect" elements, then prompt the user if they want to fix any non-right-hand elements.

**Comment by yhaomin2007** [comment]
forced non-right-hand check now, and will print the number of non-right-hand elements.

**Comment by yhaomin2007** [comment]
any further comments?

**Comment by stgeke** [comment]
@dshaver-ANL Is this ready to be merged?

**Comment by stgeke** [comment]
I think this PR should also update cgns2nek.

**Comment by yhaomin2007** [comment]
cgns2nek is also changed for v004 mesh.

**Comment by dshaver14** [comment]
one more minor change, if it doesn't detect any non-right-handed elements, it shouldn't ask the user if it wants to fix them.

**Comment by yhaomin2007** [comment]
changes are done. thanks for the comment.

File: tools/gmsh2nek/gmsh2nek.f90 [hunk 1]
```diff
@@ -1934,17 +1941,20 @@ subroutine open_re2
...
+
+      nBCre2 = 1
+      if (num_elem.ne.eftot)  nBCre2  = 2
+
 !  Write the header
       call blank     (hdr,80)    
-      write(hdr,1) num_elem, num_dim, eftot
-    1 format('#v002',i9,i3,i9,' this is the hdr')
+      write(hdr,1) num_elem, num_dim, eftot, nBCre2
+    1 format('#v004',i9,i3,i9,i3,' this is the hdr')
```
Comment (stgeke) [inline tools/gmsh2nek/gmsh2nek.f90]: I don't think the format matches the one we're using for reading.

Comment (yhaomin2007) [reply tools/gmsh2nek/gmsh2nek.f90]: fixed

**Comment by stgeke** [comment]
The format is still not correct.
See here: https://github.com/Nek5000/Nek5000/blob/master/core/reader_re2.f#L585

Please fix all mesh converters including cgns2nek.

**Comment by yhaomin2007** [comment]
write(hdr,1) num_elem, num_dim, eftot, nBCre2
    1 format('#v004',i16,i3,i16,i4,' hdr')
this is consistent with reader_re2, and also consistent with 
https://github.com/Nek5000/Nek5000/blob/11e94661faa8991b5c6e6711daf89d1cf1aa3ed5/tools/n2to3/n2to3.f#L334

I also chagned this in cgns2nek

**Review by dshaver14 — APPROVED** [review]
This looks good now. 
@stgeke Note that I had to remove the `-fallow-argument-mismatch` to get cgns2nek to compile. I think it's an incompatibility with the version of gcc I'm using (9.3.1), but a quick google search indicates that it's redundant with `-std=legacy`.

**Comment by yhaomin2007** [comment]
let me know if there is anything else you want me to change. thanks

**Pull request merged**